packages/typespec-ts/src/contextManager.ts

```diff
@@ -0,0 +1,53 @@
+import { Project } from "ts-morph";
+import { ModularMetaTree, RlcMetaTree } from "./metaTree.js";
+
+type Contexts = {
+  rlcMetaTree: RlcMetaTree;
+  modularMetaTree: ModularMetaTree;
+  outputProject: Project;
+  symbolMap: Map<string, string>;
+};
+
+type ContextKey = keyof Contexts;
+
+class ContextManager {
+  private static instance: ContextManager;
+  private contexts: Map<ContextKey, any> = new Map();
+
+  private constructor() {}
+
+  public static getInstance(): ContextManager {
+    if (!ContextManager.instance) {
+      ContextManager.instance = new ContextManager();
+    }
+    return ContextManager.instance;
+  }
+
+  public setContext<K extends ContextKey>(key: K, value: Contexts[K]): void {
+    if (this.contexts.has(key)) {
+      console.warn("Context already exists, overwriting it");
+    }
+    this.contexts.set(key, value);
+  }
+
+  public getContext<K extends ContextKey>(key: K): Contexts[K] | undefined {
```

Comment:
I think it would be nice to have some documentation for contributors on what problems the context manager is trying to solve, how to decide what should go in a context/what should become a context itself/what should just be passed by argument.

I think this context approach is a huge improvement to the codebase, I'm excited to see how we leverage it :)

Reply:
Added docs